kubeflow/fairing/builders/cluster/cluster.py

```diff
@@ -80,18 +79,31 @@ def build(self):
             ),
             spec=pod_spec
         )
-        created_pod = client. \
-            CoreV1Api(). \
-            create_namespaced_pod(self.namespace, build_pod)
+        job_spec = client.V1JobSpec(
+            template=pod_spec_template,
+            parallelism=1,
+            completions=1,
+            backoff_limit=0,
+        )
+        build_job = client.V1Job(
+            api_version="batch/v1",
+            kind="Job",
+            metadata=client.V1ObjectMeta(
+                generate_name="fairing-builder-",
+                labels=labels,
+            ),
+            spec=job_spec
+        )
+        created_job = client. \
+            BatchV1Api(). \
+            create_namespaced_job(self.namespace, build_job)
+
         self.manager.log(
-            name=created_pod.metadata.name,
-            namespace=created_pod.metadata.namespace,
+            name=created_job.metadata.name,
+            namespace=created_job.metadata.namespace,
             selectors=labels,
             container="kaniko")
 
-        # clean up created pod and secret
+        # Invoke upstream clean ups
         self.context_source.cleanup()
-        client.CoreV1Api().delete_namespaced_pod(
```

Comment:
I suggest add one option to let user choose is delete or not, by default, do not delete. for example, add a option `cleanup=False`, only user set to `True`, will delete resource, what do you think?

Reply:
Included cleanup flag.